src/stores/orderListStore.ts

```diff
@@ -54,15 +55,37 @@ class OrderListStore extends BaseStore {
   }
 
   fetchAll = async () => {
+    this.hasPendingOrders = true;
     const dto = await this.api.fetchAll();
+    this.orders = dto.map(mappers.mapToLimitOrder);
+    this.hasPendingOrders = false;
+  };
 
-    this.updateOrders(dto.map(mappers.mapToLimitOrder));
-    this.rootStore.orderBookStore.fetchAll();
+  @action
+  addOrder = (order: any) => {
+    this.orders.push(mappers.mapToLimitOrder(order));
   };
 
   @action
-  updateOrders = (orders: OrderModel[]) => {
-    this.orders = orders;
+  deleteOrder = (orderId: string) => {
```

Comment:
the implementation has the side-effect inside the predicate, plz rewrite it.
another question is why to return _boolean_? it should either act as a void or return the new array w/out order